test/integration_serial_test.go

```diff
@@ -0,0 +1,209 @@
+	out := new(bytes.Buffer)
+	os.Setenv("TF_VAR_env", "prod")
+	defer os.Unsetenv("TF_VAR_env")
+	runTerragruntRedirectOutput(t, fmt.Sprintf("terragrunt apply -auto-approve --terragrunt-non-interactive --terragrunt-working-dir %s", TEST_FIXTURE_EXTRA_ARGS_PATH), out, os.Stderr)
+	t.Log(out.String())
+	assert.Contains(t, out.String(), "Hello, World!")
+}
+
+func TestExtraArgumentsWithEnvVarBlock(t *testing.T) {
+	out := new(bytes.Buffer)
+	runTerragruntRedirectOutput(t, fmt.Sprintf("terragrunt apply -auto-approve --terragrunt-non-interactive --terragrunt-working-dir %s", TEST_FIXTURE_ENV_VARS_BLOCK_PATH), out, os.Stderr)
+	t.Log(out.String())
+	assert.Contains(t, out.String(), "I'm set in extra_arguments env_vars")
+}
+
+func TestExtraArgumentsWithRegion(t *testing.T) {
+	out := new(bytes.Buffer)
+	os.Setenv("TF_VAR_region", "us-west-2")
+	defer os.Unsetenv("TF_VAR_region")
+	runTerragruntRedirectOutput(t, fmt.Sprintf("terragrunt apply -auto-approve --terragrunt-non-interactive --terragrunt-working-dir %s", TEST_FIXTURE_EXTRA_ARGS_PATH), out, os.Stderr)
+	t.Log(out.String())
+	assert.Contains(t, out.String(), "Hello, World from Oregon!")
+}
+
+func TestPreserveEnvVarApplyAll(t *testing.T) {
+	os.Setenv("TF_VAR_seed", "from the env")
+	defer os.Unsetenv("TF_VAR_seed")
+
+	cleanupTerraformFolder(t, TEST_FIXTURE_REGRESSIONS)
+	tmpEnvPath := copyEnvironment(t, TEST_FIXTURE_REGRESSIONS)
+	rootPath := util.JoinPath(tmpEnvPath, TEST_FIXTURE_REGRESSIONS, "apply-all-envvar")
+
+	stdout := bytes.Buffer{}
+	runTerragruntRedirectOutput(t, fmt.Sprintf("terragrunt apply-all -auto-approve --terragrunt-non-interactive --terragrunt-working-dir %s", rootPath), &stdout, os.Stderr)
+	t.Log(stdout.String())
+
+	// Check the output of each child module to make sure the inputs were overridden by the env var
+	requireEnvVarModule := util.JoinPath(rootPath, "require-envvar")
+	noRequireEnvVarModule := util.JoinPath(rootPath, "no-require-envvar")
+	for _, mod := range []string{requireEnvVarModule, noRequireEnvVarModule} {
+		stdout := bytes.Buffer{}
+		err := runTerragruntCommand(t, fmt.Sprintf("terragrunt output text -no-color --terragrunt-non-interactive --terragrunt-working-dir %s", mod), &stdout, os.Stderr)
+		require.NoError(t, err)
+		assert.Contains(t, stdout.String(), "Hello from the env")
+	}
+}
+
+func TestPriorityOrderOfArgument(t *testing.T) {
+	out := new(bytes.Buffer)
+	injectedValue := "Injected-directly-by-argument"
+	runTerragruntRedirectOutput(t, fmt.Sprintf("terragrunt apply -auto-approve -var extra_var=%s --terragrunt-non-interactive --terragrunt-working-dir %s", injectedValue, TEST_FIXTURE_EXTRA_ARGS_PATH), out, os.Stderr)
+	t.Log(out.String())
+	// And the result value for test should be the injected variable since the injected arguments are injected before the suplied parameters,
+	// so our override of extra_var should be the last argument.
+	assert.Contains(t, out.String(), fmt.Sprintf(`test = "%s"`, injectedValue))
+}
+
```

Comment:
The two tests after this line are new. Everything before are from `integration_test.go` that needed to run in serial.